cfme/cloud/instance/__init__.py

```diff
@@ -465,26 +526,93 @@ def resetter(self, *args, **kwargs):
         self.view.toolbar.reload.click()
 
 
+@navigator.register(Instance, 'DetachVolume')
+class DetachVolume(CFMENavigateStep):
+    VIEW = DetachVolumeView
+
+    def prerequisite(self, *args, **kwargs):
+        if not self.obj.provider.one_of(OpenStackProvider):
+            raise DestinationNotFound('Detach Volume only available for Openstack instances')
+        elif self.obj.appliance.version < '5.7':
+            raise DestinationNotFound('Detach Volume not available for appliance version')
+        return navigate_to(self.obj, 'Details')
+
+    def step(self):
+        try:
+            self.prerequisite_view.toolbar.configuration.item_select(
+                'Detach a Cloud Volume from this Instance')
+        except NoSuchElementException:
+            raise DestinationNotFound('Detach Cloud Volume option not available for instance')
+
+
 @navigator.register(Instance, 'Edit')
 class Edit(CFMENavigateStep):
+    VIEW = EditView
     prerequisite = NavigateToSibling('Details')
 
-    # No am_i_here because the page only indicates name and not provider
-
     def step(self, *args, **kwargs):
         self.prerequisite_view.toolbar.configuration.item_select('Edit this Instance')
 
 
-@navigator.register(Instance, 'EditMgmtEngineRelation')
-class EditMgmtEngineRelation(CFMENavigateStep):
+@navigator.register(Instance, 'EditManagementEngineRelationship')
```

Comment:
my bad, I thought destination name "EditManagementEngineRelationship" would be tooooo long
I hope you have renamed these destinations in my tests :)